pkg/transforms/policy.go

```diff
@@ -37,6 +41,106 @@ func PolicyResourceBuilder(p *p.Policy) *PolicyResource {
+// If `parentPolicyNs`, `parentPolicyName` do not exist, the policy is a stand-alone policy
+func getPolicyCommonProperties(c *unstructured.Unstructured, node Node) Node {
+	annos := c.GetAnnotations()
+	// This annotation is large and useless
+	delete(annos, "kubectl.kubernetes.io/last-applied-configuration")
+
+	node.Properties["annotations"] = annos
+
+	node.Properties["isExternal"] = false
+
+	for _, m := range c.GetManagedFields() {
+		if m.Manager == "multicluster-operators-subscription" ||
+			strings.Contains(m.Manager, "argocd") {
+			node.Properties["isExternal"] = true
+		}
+	}
+
+	typeMeta := metav1.TypeMeta{
+		Kind:       c.GetKind(),
+		APIVersion: c.GetAPIVersion(),
+	}
+
+	apiGroupVersion(typeMeta, &node) // add kind, apigroup and version
+
+	policyNsName, ok := c.GetLabels()["policy.open-cluster-management.io/policy"]
+	if ok {
+		stringSlice := strings.Split(policyNsName, ".")
+		// If len(stringSlice) != 2, failed to parse policyNsName
+		if len(stringSlice) == 2 {
+			node.Properties["parentPolicyNs"] = stringSlice[0]
+			node.Properties["parentPolicyName"] = stringSlice[1]
+		}
+	}
+
+	node.Properties["compliant"], _, _ = unstructured.NestedString(c.Object, "status", "compliant")
+
+	node.Properties["severity"], _, _ = unstructured.NestedString(c.Object, "spec", "severity")
+
+	node.Properties["remediationAction"], _, _ = unstructured.NestedString(c.Object, "spec", "remediationAction")
+
+	return node
+}
+
+// For cert, config policies.
+func StandalonePolicyResourceBuilder(c *unstructured.Unstructured) *PolicyResource {
+	node := transformCommon(c) // Start off with the common properties
+
+	return &PolicyResource{node: getPolicyCommonProperties(c, node)}
+}
+
+func OperatorPolicyResourceBuilder(c *unstructured.Unstructured) *PolicyResource {
+	node := transformCommon(c) // Start off with the common properties
+
+	node = getPolicyCommonProperties(c, node)
+
+	var deploymentAvailable string
+	var upgradeAvailable string
+
+	conditions, _, _ := unstructured.NestedSlice(c.Object, "status", "conditions")
+	for _, condition := range conditions {
+		mapCondition, ok := condition.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		conditionType, found, err := unstructured.NestedString(mapCondition, "type")
+		if !found || err != nil {
+			continue
+		}
+
+		conditionReason, found, err := unstructured.NestedString(mapCondition, "reason")
+		if !found || err != nil {
+			continue
+		}
+
+		if conditionType == "InstallPlanCompliant" {
+			if conditionReason == "InstallPlanRequiresApproval" || conditionReason == "InstallPlanApproved" {
+				upgradeAvailable = "yes"
+			} else {
+				upgradeAvailable = "no"
+			}
+		} else if conditionType == "DeploymentCompliant" {
+			if conditionReason == "DeploymentsAvailable" {
+				deploymentAvailable = "yes"
+			} else {
+				deploymentAvailable = "no"
+			}
+		}
+	}
+
+	node.Properties["deploymentAvailable"] = deploymentAvailable
+	node.Properties["upgradeAvailable"] = upgradeAvailable
```

Comment:
@mprahl Could the OperatorPolicies crd define these in `additionalPrinterColumns`? Then we won't need this after your other change.

Reply:
I think so. I'll look into this later.